runners/spark/src/main/java/org/apache/beam/runners/spark/metrics/SparkBeamMetric.java

```diff
@@ -17,27 +17,30 @@
  */
 package org.apache.beam.runners.spark.metrics;
 
+import static java.util.stream.Collectors.toList;
 import static org.apache.beam.runners.core.metrics.MetricsContainerStepMap.asAttemptedOnlyMetricResults;
 
 import com.codahale.metrics.Metric;
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Map;
 import org.apache.beam.runners.core.metrics.MetricsContainerStepMap;
 import org.apache.beam.sdk.metrics.DistributionResult;
 import org.apache.beam.sdk.metrics.GaugeResult;
+import org.apache.beam.sdk.metrics.MetricKey;
 import org.apache.beam.sdk.metrics.MetricName;
 import org.apache.beam.sdk.metrics.MetricQueryResults;
 import org.apache.beam.sdk.metrics.MetricResult;
 import org.apache.beam.sdk.metrics.MetricResults;
 import org.apache.beam.sdk.metrics.MetricsFilter;
 import org.apache.beam.vendor.guava.v20_0.com.google.common.annotations.VisibleForTesting;
+import org.apache.beam.vendor.guava.v20_0.com.google.common.collect.ImmutableList;
 
 /**
  * An adapter between the {@link MetricsContainerStepMap} and Codahale's {@link Metric} interface.
  */
 class SparkBeamMetric implements Metric {
```

Comment:
I also notice the Metric class will only handle User Metrics (Name, Namespace). If spark only cares about UserMetrics that's okay I suppose. but won't handle more generlized MonitoringInfos

Reply:
Agreed. Most of what I've done so far (in #7823) is [expanding the Java SDK metrics classes to support the wider MonitoringInfo data model] while [preserving semantics of existing code that assumes / requires user-metrics].

In this case, [I end up adapting this](https://github.com/apache/beam/pull/7823/files#diff-2b144bc5b4cec16779fa6126425c6fe3R71) to use [a `MetricLabels.value()` API](https://github.com/apache/beam/pull/7823/files#diff-245a1fd7e743b2f382c978a12028f8c2R46) that assumes there is exactly one label set on the MonitoringInfo (`PTRANSFORM` xor `PCOLLECTION`) and uses that, so it should be reasonably "future-proofed"